halo2_backend/src/plonk/prover.rs

```diff
@@ -114,16 +116,22 @@ pub struct ProverV2<
 > {
     // Circuit and setup fields
     params: &'params Scheme::ParamsProver,
+    // Plonk proving key
```

Comment:
Maybe rename this to `Halo2 proving key` or `Halo2 proof system proving key`?

Reply:
I see, I think you have a point.  After all, there's a folder in halo2 with the proving system implementation that is called `plonk`.  Nevertheless I find this confusing :confounded:

We have the plonk protocol, and the halo2 protocol which extends the plonk protocol.
Then we have the halo2 library which implements the halo2 protocol.
And then we have the halo2 library that offers an API to write plonkish circuits to be used with the halo2 backend that implements the halo2 protocol.
And all this follows a library interface for plonkish proof systems defined in halo2_middleware.

I don't know how we can solve this; other than trying to be specific when we say halo2! (halo2 proof system, halo2 circuit, halo2 frontend library, halo2 backend library, etc).

Anyway, this is outside the topic of my original comment :stuck_out_tongue_closed_eyes: 

Since the proof system is implemented in the `plonk` folder, I think it's OK to call the pk `Plonk proving key`, so I'll mark this as resolved!